### ISSUE TITLE
Complete quoting for parameters of some CMake commands

### ISSUE THREAD

**Issue by elfring** [SFML/CSFML#33]
Some parameters (like "[${CMAKE_CURRENT_SOURCE_DIR}/include](https://github.com/LaurentGomila/CSFML/blob/6669a6590a59c2fa2ccaed018ff42cc0ec2ebb6d/CMakeLists.txt#L28)" and "[${DOXYGEN_OUTPUT_DIR}/doxyfile](https://github.com/LaurentGomila/CSFML/blob/6669a6590a59c2fa2ccaed018ff42cc0ec2ebb6d/doc/CMakeLists.txt#L26)") are passed to CMake commands in your build scripts without enclosing them by quotation marks. I see that these places will result in build difficulties if the contents of the used variables will contain special characters like spaces.

I would recommend to apply [advices from a Wiki article](http://cmake.org/Wiki/CMake/Language_Syntax#CMake_splits_arguments_unless_you_use_quotation_marks_or_escapes. "CMake splits arguments unless you use quotation marks or escapes.").

**Comment by zsbzsb** [comment]
@elfring I know its a bit late to respond to this, but are you sure this is an actual issue? I have built CSFML in directories that contain spaces and never had an issue. And just to be clear, SFML also doesn't put quotes around the paths and I have never heard of one person complaining. [See here](https://github.com/SFML/SFML/blob/master/CMakeLists.txt#L46) for reference.

@LaurentGomila You have more experience with CMake, could this possibly be an issue (and would it also have to be corrected in SFML)?

**Comment by LaurentGomila** [comment]
Since I'm not 100% sure, let's wait until it actually becomes an issue before changing everything.

**Comment by zsbzsb** [comment]
Alright, then I am going to close this as it will most likely become an issue in SFML before CSFML and if it does we can port the changes.

**Comment by elfring** [comment]
Do you get [any further ideas from an other bug report](https://cmake.org/Bug/view.php?id=15076 "Check handling of semicolons")?

**Comment by zsbzsb** [comment]
I don't understand you... I did some looking around and you did you posted the same issue [on the SFML tracker #534](https://github.com/SFML/SFML/issues/534). @Bromeon gave you basically the same answer I did that this has never been an issue - then you respond with some weird scenario about semicolons being part of the path. Unless this actually causes issues in real scenarios (who /really/ uses semicolons in a directory name?) its not an issue.

One other thing, [it looks like you went and copy-pasted this same "issue" to countless repos](https://www.google.com/search?q=elfring+cmake+quoting&ie=utf-8&oe=utf-8). And considering that CMake hasn't even fixed it yet in their core scripts I really doubt this of any real importance.
